src/core/message.ts

```diff
@@ -288,4 +298,20 @@ export class DefaultMessage implements Message {
   equal(message: Message): boolean {
     return this.serial === message.serial;
   }
+
+  with(event: MessageEventPayload): Message {
+    if (event.type === MessageEvents.Created) {
+      throw new ErrorInfo('Cannot apply a created event to a message', 40000, 400);
+    }
+
+    if (event.message.serial !== this.serial) {
+      throw new ErrorInfo('Cannot apply event for a different message', 40000, 400);
+    }
+
+    if (this.version >= event.message.version) {
+      return this;
+    }
+
+    return event.message;
```

Comment:
```suggestion
    return this.version >= event.message.version ? this : event.message
```

Ternary perhaps a bit cleaner?

Reply:
Done